compiler/src/scala/quoted/runtime/impl/printers/SourceCode.scala

```diff
@@ -809,13 +809,10 @@ object SourceCode {
         case Nil => Flags.EmptyFlags
         case arg :: _ => arg.symbol.flags
       }
-      if (argFlags.is(Flags.Erased | Flags.Given)) {
-        if (argFlags.is(Flags.Given)) this += " given"
-        if (argFlags.is(Flags.Erased)) this += " erased"
-        this += " "
-      }
       inParens {
         if (argFlags.is(Flags.Implicit) && !argFlags.is(Flags.Given)) this += "implicit "
+        if (argFlags.is(Flags.Given)) this += "using "
```

Comment:
I'm not entirely sure to understand this one. Are you saying that if the arg flag includes given but the outer type is a lambda then the keyword `using` is invalid because it should be a contextual function type ?

Reply:
Oh I see, you meant that the test case failed. I should have checked CI before asking the question above. Will take a look.

Reply:
0910940e2369a1fe20067d694c2d0f817579bd92 is an attempt to ignore lambdas, although it feels very targeted to me. Let me know what you think of it.

Reply:
Maybe it would be better to split into `printMethdArgsDefs` and `printLambdaArgsDefs` as those are different grammar.

Reply:
Addressed in 34c7058. Let me know if that seems more readable to you or not.

Reply:
> Addressed in [34c7058](https://github.com/lampepfl/dotty/pull/17952/commits/34c705891821df0bb4e3f93a726add03d5dc2c1e). Let me know if that seems more readable to you or not.

That seems fine to me, but I would deduplicate the two `printSeparated` local defs into one private def reused by both methods.

Reply:
Done via https://github.com/lampepfl/dotty/pull/17952/commits/25f78eacd038fcb7902f6ffa261bbe004c2157db.

Reply:
We can simplify this to 
```diff
- if (argFlags.is(Flags.Implicit) && !argFlags.is(Flags.Given)) this += "implicit "
- if (argFlags.is(Flags.Given)) this += "using "
+ if argFlags.is(Flags.Given) then this += "using "
+ else if argFlags.is(Flags.Implicit) then this += "implicit "